tests/pipelines/test_pipelines_common.py

```diff
@@ -143,6 +143,18 @@ def gen_test(ModelClass, checkpoint, tiny_config, tokenizer_class, feature_extra
             @skipIf(tiny_config is None, "TinyConfig does not exist")
             @skipIf(checkpoint is None, "checkpoint does not exist")
             def test(self):
+                if tokenizer_class is not None:
+                    try:
+                        tokenizer = get_tiny_tokenizer_from_checkpoint(checkpoint)
+                        tiny_config.vocab_size = len(tokenizer)
```

Comment:
I needed to modify the basis of the pipeline tests as they use the `train_new_from_iterator` feature to create toy tokenizers.

The problem that arose was the difference between the default `tiny_config.vocab_size` attribute and the size of the toy tokenizer vocabulary. With my modification, the vocabulary is larger and exceeded this default value. 

So I reordered the lines (to create the tokenizer before the model) and added this line.

Reply:
Thanks a lot for your review @Narsil :hugs: !

> How many tests were impacted

80 tests failed in `run_tests_pipelines_tf` CI and 132 in `run_tests_pipelines_torch` CI. I've observed that it impacted 11 different model configurations ('Bart', 'Blenderbot',  'Deberta',  'GPT2',  'GPTJ',  'GPTNeo',  'IBert',  'LED',  'Longformer',  'Roberta' and  'Yoso')


> The other fix I can see would be instead to modify ModelTester.get_pipeline_config to add the necessary amount of vocabulary_size so that at least we have enough vocab to be able to retrain the tokenizer.
wdyt ?

I really like this suggestion! I made the changes in the last commits. Can you tell me if you're ok with those? I put a vocabulary size of 300 so that it would be rounded up to the nearest hundred above 256.

Reply:
Thanks for this ! 

I really think this is better this way ! Thanks for taking care of it .